packages/react-topology/src/behavior/withCreateConnector.tsx

```diff
@@ -37,8 +37,15 @@ type CreateConnectorWidgetProps = {
     element: Node,
     target: Node | Graph,
     event: DragEvent,
+    dropHints?: string[] | undefined,
     choice?: ConnectorChoice
-  ) => ConnectorChoice[] | void | undefined | null | React.ReactElement[];
+  ) =>
+    | Promise<ConnectorChoice[] | void | undefined | null | React.ReactElement[]>
+    | ConnectorChoice[]
+    | void
+    | undefined
+    | null
+    | React.ReactElement[];
```

Comment:
Add new type. No idea why we originally had `React.ReactElement[]` in the return type, so let's remove that as well.
```
type OnCreateResult = ConnectorChoice[] | void | undefined | null;
```

```suggestion
  ) => Promise<OnCreateResult> | OnCreateResult;
```